node/src/neighborhood/node_location.rs

```diff
@@ -0,0 +1,208 @@
+    ) {
+        let mut node = make_node_record(2222, true);
+        let country_record = pick_country_code_record(2222);
+        println!("node: {:?}", node);
+        node.metadata.node_location_opt = Some(NodeLocation {
+            country_code: country_record.1.clone(),
+            free_world_bit: country_record.2,
+        });
+        node.metadata.node_addr_opt = Some(NodeAddr::new(&country_record.0, &[8000 % 10000]));
+        node.inner.country_code = country_record.1;
+        let db = db_from_node(&node);
+        let builder = GossipBuilder::new(&db);
+
+        let builder = builder.node(node.public_key(), true);
+
+        let mut gossip = builder.build();
+        let gossip_result = gossip.node_records.remove(0);
+        let node_record = NodeRecord::try_from(&gossip_result).unwrap();
+
+        println!("node_record: {:?}", node_record);
+        assert_eq!(
+            gossip_result.node_addr_opt.unwrap(),
+            node.node_addr_opt().unwrap()
+        );
+        assert_eq!(node_record.inner.country_code, "US")
+    }
+}
+
+#[allow(dead_code, unused_imports)]
+#[cfg(not(test))]
+mod test_ip_country_performance {
+    use crate::neighborhood::node_location::{get_node_location, NodeLocation};
+    use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+    use std::time::SystemTime;
+
+    #[test]
+    fn get_node_location_for_test_with_1000_v4_high_val_ips() {
+        let mut result_vec: Vec<Option<NodeLocation>> = vec![];
+        let start = 0xdfff8000u32;
+        let timestart = SystemTime::now();
+
+        (start..(start + 1000)).for_each(|num| {
+            let address = IpAddr::V4(Ipv4Addr::from(num));
+            result_vec.push(get_node_location(Some(address)));
+        });
+
+        let timeend = SystemTime::now();
+
+        println!(
+            "get_node_location_for_test_with_1000_v4_high_val_ips Elapesd time: {}",
+            timeend.duration_since(timestart).unwrap().as_secs()
+        );
+        assert!(timeend.duration_since(timestart).unwrap().as_secs() < 20);
+        while !result_vec.is_empty() {
+            let location = result_vec.remove(0);
+            assert_eq!(location.unwrap().free_world_bit, true);
+        }
+    }
+
+    #[test]
+    fn get_node_location_for_test_with_1000_v4_low_val_ips() {
+        let mut result_vec: Vec<Option<NodeLocation>> = vec![];
+        let start = 0x10100101u32;
+        let timestart = SystemTime::now();
+
+        (start..(start + 1000)).for_each(|num| {
+            let address = IpAddr::V4(Ipv4Addr::from(num));
+            result_vec.push(get_node_location(Some(address)));
+        });
+
+        let timeend = SystemTime::now();
+
+        println!(
+            "get_node_location_for_test_with_1000_v4_high_val_ips Elapesd time: {}",
+            timeend.duration_since(timestart).unwrap().as_secs()
+        );
+        assert!(timeend.duration_since(timestart).unwrap().as_secs() < 20);
```

Comment:
_I still think we might end up deleting these tests but:_

It doesn't feel right that you set the hard limit as bug as for the other test aimed at IP addresses on the other end of the spectrum. We don't know how much it should take but we do know that there should be significant difference in these different searched areas. I think it should be reflected by the test, so that the boundaries are not identically generous in how much performance time you are willing to consider acceptable.